samples/dfu/src/main.c

```diff
@@ -243,6 +244,9 @@ void main(void)
 		}
 	}
 
+	if (IS_ENABLED(CONFIG_GOLIOTH_SAMPLE_ETHERNET)) {
+		eth_connect();
+	}
 	if (IS_ENABLED(CONFIG_GOLIOTH_SAMPLE_WIFI)) {
 		LOG_INF("Connecting to WiFi");
 		wifi_connect();
```

Comment:
```suggestion
	if (IS_ENABLED(CONFIG_GOLIOTH_SAMPLE_WIFI)) {
		LOG_INF("Connecting to WiFi");
		wifi_connect();
	}
	
	if (IS_ENABLED(CONFIG_GOLIOTH_SAMPLE_DHCP_START)) {
		net_dhcpv4_start(net_if_get_default());
```

If we put that after "WiFi connect" then in the future we might handle the case when DHCP needs to be started for WiFi network. This might be needed for more "natve" WiFi support, like ESP32 (Espressif is working on adjusting to Zephyr WiFi APIs) or nRF WiFi.